rpc_proxy.go

```diff
@@ -199,7 +263,29 @@ func (p *rpcProxy) director(ctx context.Context,
 
 	outCtx := metadata.NewOutgoingContext(ctx, mdCopy)
 
-	return outCtx, p.lndConn, nil
+	// Direct the call to the correct backend. For lnd we _always_ have a
```

Comment:
I might be missing some context here on why this is needed/done this way, but why wouldn't we unify this such that all backends were treated the same (lnd, loop, pool, faraday)?

Reply:
It's because of the grpc-web proxy. We need to create our own `grpc.Server` instance with the custom codec enabled. We can hook the "guest" daemons into that one directly because we added the `StartAsSubserver()` method to them. The gRPC server in `lnd` is a black box to us and that's why we always have to forward the call to a connection, even if that connection is to a port in the same process.

We could probably simplify this by exposing the RPC server in `lnd`. But for this PR it actually made it way easier to add the remote modes of the other daemons because this was already in place.

Reply:
Ah, of course. Would it be an idea to add the `StartAsSubserver` option to lnd going forward? Can easily be done together with the `grpc` refactoring I'm doing for lnd (I think).

Reply:
Yes, sure. That would make it a bit nicer. And LiT would actually become the main process and not just a barnacle around `lnd` :joy: